src/Hooks/Dispatchers/ReplaceCommandHookDispatcher.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Consolidation\AnnotatedCommand\Hooks\Dispatchers;
+
+use Consolidation\AnnotatedCommand\CommandData;
+use Consolidation\AnnotatedCommand\Hooks\HookManager;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleCommandEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Call hooks.
+ */
+class ReplaceCommandHookDispatcher extends HookDispatcher
+{
+
+    /**
+     * @return int
+     */
+    public function hasReplaceCommandHook()
+    {
+        return count($this->getReplaceCommandHooks());
+    }
+
+    /**
+     * @return \callable[]
+     */
+    public function getReplaceCommandHooks()
+    {
+        $hooks = [
+            HookManager::REPLACE_COMMAND_HOOK,
+        ];
+        $replaceCommandHooks = $this->getHooks($hooks);
+
+        return $replaceCommandHooks;
+    }
+
+    /**
+     * @param \Consolidation\AnnotatedCommand\CommandData $commandData
+     *
+     * @return callable
+     */
+    public function getReplacementCommand(CommandData $commandData)
+    {
+        $replaceCommandHooks = $this->getReplaceCommandHooks($commandData);
+
+        // We only take the first hook implementation of "replace-command" as the replacement. Commands shouldn't have
+        // more than one replacement.
```

Comment:
Yes, I did mean CommandProcessor, but I don't know what I was looking at that made me think there was a logger there, as none of the classes in this project log at the moment.

I don't want `$output` references here. To fix this, use [LoggerAwareTrait](https://github.com/php-fig/log/blob/master/Psr/Log/LoggerAwareTrait.php) (and LoggerAwareInterface, in the same project). You'd have to add it to CommandProcessor, and have CommandProcessor set the logger on the ReplaceCommandHookDispatcher when it was instantiated. You'd need to silently not log if no logger is provided (for backwards compatibility).

However, since there is no logging at all in this project ATM, it's also okay with me if you just back out the `$output` change, and leave logging out for now.